site/guides/enhanced-apis.md

```diff
@@ -0,0 +1,86 @@
+---
+outline: deep
+head:
+  - - meta
+    - property: og:title
+      content: How to Use Alchemy Enhanced APIs
+  - - meta
+    - name: description
+      content: Follow this guide to use Alchemy Enhanced APIs with Account Kit, a vertically integrated stack for building apps that support ERC-4337.
+  - - meta
+    - property: og:description
+      content: Follow this guide to use Alchemy Enhanced APIs with Account Kit, a vertically integrated stack for building apps that support ERC-4337.
+  - - meta
+    - name: twitter:title
+      content: How to Use Alchemy Enhanced APIs
+  - - meta
+    - name: twitter:description
+      content: Follow this guide to use Alchemy Enhanced APIs with Account Kit, a vertically integrated stack for building apps that support ERC-4337.
+---
+
+# How to Use Alchemy Enhanced APIs
+
+Alchemy provides several [Enhanced APIs](https://www.alchemy.com/enhanced-apis) to make it easier for web3 developers to query data from blockchains. These are especially useful for querying information about the smart accounts you create using Account Kit, such as the account's token balances using the [Token API](https://www.alchemy.com/token-api) or the account's owned NFTs using the [NFT API](https://www.alchemy.com/nft-api).
+
+There are two ways you can use Alchemy Enhanced APIs using Account Kit:
+
+1. via the [Alchemy SDK](https://www.alchemy.com/sdk)
+2. via a custom [Viem Client](https://viem.sh/docs/clients/custom.html)
+
+For the purpose of this example, we'll stick to understanding how to use either approach to leverage two of Alchemy's Enhanced APIs: the Token API and NFT API.
+
+## 1. Alchemy SDK
+
+Alchemy has developed a Typescript SDK to make development with the Enhanced APIs simple. The SDK includes ways to leverage Alchemy's Simulation API, Token API, Transact API, NFT API, Webhooks and Websockets, and more across Alchemy's supported chains. You can take a look at the code [here](https://github.com/alchemyplatform/alchemy-sdk-js).
+
+To use the Alchemy SDK in your project directory, you'll need to install the required packages:
+
+::: code-group
+
+```bash [npm]
+npm i alchemy-sdk
+```
+
+```bash [yarn]
+yarn add alchemy-sdk
+```
+
+:::
+
+Then, it's simply a matter of creating an `AlchemyProvider` from Account Kit and then creating an `alchemy` client from the Alchemy SDK. For example, below we can get the smart account's address from the `AlchemyProvider` in order query the smart account's token and NFT balances in just 1 line of code each!
+
+<<< @/snippets/enhanced-apis-example/alchemy-sdk/example.ts
+
+There are so many more Enhanced APIs the the Alchemy SDK exposes, and can be useful for development with Account Abstraction. Try it out [here](https://github.com/alchemyplatform/alchemy-sdk-js)!
+
+## 2. Viem Client
```

Comment:
This flow is really complicated imo

I think we should:
1) add alchemy-sdk as a peer dependency
2) create a method on the `AlchemyProvider` called `withAlchemyEnhancedApis` that does the following:
```ts
withAlchemyEnhancedApis(): this & Alchemy {
    return this.extend((provider) => {
      const alchemy = new Alchemy({
        url: provider.rpcClient.transport.url,
      });

      return {
        core: alchemy.core,
        nft: alchemy.nft,
        transact: alchemy.transact,
        debug: alchemy.debug,
        ws: alchemy.ws,
        notify: alchemy.notify,
        config: alchemy.config,
      };
    });
  }

This would simplify this guide massively + improve the developer experience a lot if they wanted to do this. Because it's a peer dependency it should also minimize bundle size since it's now opt-in for devs
```